### PR TITLE
Issue #3368689 by tbsiqueira: Update behat tests to use the new compatible php 8.1 images

### PR DESCRIPTION
**Pull request by tbsiqueira**
## :rocket: :fire: Related PRs :fire: :rocket:

This is a PR from a series of 5 PRs that should be merged in the following order: 

1. https://github.com/goalgorilla/open_social_dev/pull/70 
2. https://github.com/goalgorilla/open_social/pull/3460
3. https://github.com/goalgorilla/open_social/pull/3457
4. https://github.com/goalgorilla/drupal_social/pull/392 ( this is just a private repository that we use for development, this will not be available for public users )
5. https://github.com/goalgorilla/open_social/pull/3439

## Problem
We are now officially adding php 8.1 support, but our behat tests are still running on PHP 8.0, this is not an issue because PHPStan guarantees the compatibility, but we want to run our behat tests using PHP 8.1,

After we fixed the behat tests, a couple of issues raised:

`$extensions = explode(' ', mb_strtolower(trim($config->get('download_count_excluded_file_extensions') ?? "")));`

Unresolved variation cache module dependencies during hook updates fixed here and other places, this errors were found during the upgrade path check, and since they are green, they should all be resolved now.

```
 // Add variationcache hook update from group as required to fix update path.
  $dependencies['social_core'][11402] = [
    'group' => 8020,
  ];

  // Add variationcache hook update from group as required to fix update path.
  $dependencies['social_core'][11900] = [
    'group' => 8020,
  ];
```

Missing UUID on configuration created during hook updates, as a result of not using the correct Drupal APIs to generate configs from yml files, [as seen here](https://github.com/goalgorilla/open_social/pull/3439/files#diff-e71c5f38a4b8eb21a16ec516bda780831bf47946fb56a5ea6130e919231a6ae6L1564): 

```
  $config_storage = \Drupal::service('config.storage');
  $config_path = \Drupal::service('extension.list.module')->getPath('social_core') . '/config/static';
  $source = new FileStorage($config_path);
  $config_storage->write('core.entity_view_mode.node.email_card', (array) $source->read('core.entity_view_mode.node.email_card_11401'));
```

This last issue required an update hook, to create a proper for those configuration files, [change is available at this link](https://github.com/goalgorilla/open_social/pull/3439/files#diff-76261ef8588b9c0e238662e1597bf7c1dc0d0db483c3e21fd3fb897e8feb6976R2260)

I also noticed there some `explode()` functions were .... blowing up :boom: ( pun intended ) due to the "string" argument being NULL, so I added checks to avoid that, and they are all verified with PHPStan, the ones that don't have it is because we don't need.

## Solution
Update docker images used for Behat tests to feature PHP 8.1 compatible images

## Issue tracker
https://www.drupal.org/project/social/issues/3368689

## Theme issue tracker
N/A

## How to test
- [ ] Green tests

### Before merge
- [x] Code/peer review is completed
- [ ] All commit messages are [clear and clean](https://open-social.slite.com/app/docs/DnmermZDIx_0OQ). If applicable a rebase was performed
- [ ] All automated tests are green
- [ ] Functional/manual tests of the acceptance criteria are approved
- [ ] All acceptance criteria were met
- [ ] New features or changes to existing features are covered by tests, either unit (preferably) or behat
- [ ] Update path is tested. New hook_updates should respect update order, right naming convention and consider hook_post_update code
- [ ] Module can be safely uninstalled. Update/implement hook_uninstall and make sure that removed configuration or dependencies are removed/uninstalled
- [ ] This pull request has all required labels (team/type/priority)
- [ ] This pull request has a milestone
- [ ] This pull request has an assignee (if applicable)
- [ ] Any front end changes are tested on all major browsers
- [ ] New UI elements, or changes on UI elements are approved by the design team
- [ ] New features, or feature changes are approved by the product owner

### After merge
- [ ] Code is tested on all branches that it has been cherry-picked
- [ ] Update hook number might need adjustment, make sure they have the correct order
- [ ] The Drupal.org ticket(s) are updated according to this pull request status

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3439-lbpxakgzwrqrve1bw7r81n4ktuajcudc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64998ad2757563ca37f7e7df

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3439-lbpxakgzwrqrve1bw7r81n4ktuajcudc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64998ad2757563ca37f7e7df

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3439-lbpxakgzwrqrve1bw7r81n4ktuajcudc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64998ad2757563ca37f7e7df

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3439-lbpxakgzwrqrve1bw7r81n4ktuajcudc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64998ad2757563ca37f7e7df

**Comment by open-social-tugboat** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3439-lbpxakgzwrqrve1bw7r81n4ktuajcudc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64998ad2757563ca37f7e7df

**Comment by Kingdutch** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3439-lbpxakgzwrqrve1bw7r81n4ktuajcudc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64998ad2757563ca37f7e7df

**Comment by Kingdutch** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3439-lbpxakgzwrqrve1bw7r81n4ktuajcudc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64998ad2757563ca37f7e7df

**Comment by Kingdutch** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3439-lbpxakgzwrqrve1bw7r81n4ktuajcudc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64998ad2757563ca37f7e7df

**Comment by Kingdutch** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3439-lbpxakgzwrqrve1bw7r81n4ktuajcudc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64998ad2757563ca37f7e7df

**Comment by Kingdutch** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3439-lbpxakgzwrqrve1bw7r81n4ktuajcudc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64998ad2757563ca37f7e7df

**Comment by Kingdutch** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3439-lbpxakgzwrqrve1bw7r81n4ktuajcudc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64998ad2757563ca37f7e7df

**Comment by Kingdutch** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3439-lbpxakgzwrqrve1bw7r81n4ktuajcudc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64998ad2757563ca37f7e7df

**Comment by Kingdutch** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3439-lbpxakgzwrqrve1bw7r81n4ktuajcudc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64998ad2757563ca37f7e7df

**Comment by Kingdutch** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3439-lbpxakgzwrqrve1bw7r81n4ktuajcudc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64998ad2757563ca37f7e7df

**Comment by Kingdutch** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3439-lbpxakgzwrqrve1bw7r81n4ktuajcudc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64998ad2757563ca37f7e7df

**Comment by Kingdutch** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3439-lbpxakgzwrqrve1bw7r81n4ktuajcudc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64998ad2757563ca37f7e7df

**Comment by Kingdutch** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3439-lbpxakgzwrqrve1bw7r81n4ktuajcudc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64998ad2757563ca37f7e7df

**Comment by Kingdutch** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3439-lbpxakgzwrqrve1bw7r81n4ktuajcudc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64998ad2757563ca37f7e7df

**Comment by tbsiqueira** [comment]
This has been supressed by

**Comment by Kingdutch** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3439-lbpxakgzwrqrve1bw7r81n4ktuajcudc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64998ad2757563ca37f7e7df

**Comment by Kingdutch** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3439-lbpxakgzwrqrve1bw7r81n4ktuajcudc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64998ad2757563ca37f7e7df

**Comment by Kingdutch** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3439-lbpxakgzwrqrve1bw7r81n4ktuajcudc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64998ad2757563ca37f7e7df

**Comment by Kingdutch** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3439-lbpxakgzwrqrve1bw7r81n4ktuajcudc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64998ad2757563ca37f7e7df

**Comment by Kingdutch** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3439-lbpxakgzwrqrve1bw7r81n4ktuajcudc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64998ad2757563ca37f7e7df

File: modules/social_features/social_core/social_core.install [hunk 1]
```diff
@@ -1561,11 +1563,19 @@ function social_core_update_11301(): void {
...
+
+    try {
+      $content = file_get_contents(\Drupal::service('extension.list.module')->getPath('social_core') . '/config/static/core.entity_view_mode.node.email_card_11401.yml');
+      if (is_string($content)) {
+        $yml = Yaml::parse($content);
+        EntityViewMode::create($yml)->save();
+      }
+    }
+    catch (EntityStorageException $e) {
+      \Drupal::logger('social_event')->error($e->getMessage());
```
Comment (ronaldtebrake) [inline modules/social_features/social_core/social_core.install:1575]: `\Drupal::logger('social_core')->error($e->getMessage());`

Comment (tbsiqueira) [reply modules/social_features/social_core/social_core.install:1575]: Changed ;D

File: modules/social_features/social_event/social_event.install [hunk 2]
```diff
@@ -2111,29 +2117,36 @@ function social_event_update_10311(): string {
...
+function social_event_update_11401() {
 
   $config_storage = \Drupal::service('config.storage');
   $config_path = \Drupal::service('extension.list.module')->getPath('social_event') . '/config/static';
   $source = new FileStorage($config_path);
 
-  foreach ($config_files as $origin_name => $update_name) {
-    $config_storage->write($origin_name, (array) $source->read($update_name));
-  }
+  $config_storage->write('message.template.user_was_enrolled_to_event', (array) $source->read('message.template.user_was_enrolled_to_event_11401'));
```
Comment (ronaldtebrake) [inline modules/social_features/social_event/social_event.install:2126]: this can be removed right?

Comment (tbsiqueira) [reply modules/social_features/social_event/social_event.install:2126]: No because this needs to be created

Comment (ronaldtebrake) [reply modules/social_features/social_event/social_event.install:2126]: Wasn't the entire purpose of this to use the API though? and thus remove the config storage and rather use entity api related functions?

Comment (tbsiqueira) [reply modules/social_features/social_event/social_event.install:2126]: Done

**Comment by Kingdutch** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3439-lbpxakgzwrqrve1bw7r81n4ktuajcudc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64998ad2757563ca37f7e7df

**Comment by Kingdutch** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3439-lbpxakgzwrqrve1bw7r81n4ktuajcudc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64998ad2757563ca37f7e7df

**Comment by Kingdutch** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3439-lbpxakgzwrqrve1bw7r81n4ktuajcudc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64998ad2757563ca37f7e7df

**Comment by Kingdutch** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3439-lbpxakgzwrqrve1bw7r81n4ktuajcudc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64998ad2757563ca37f7e7df

**Comment by Kingdutch** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3439-lbpxakgzwrqrve1bw7r81n4ktuajcudc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64998ad2757563ca37f7e7df

**Comment by Kingdutch** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3439-lbpxakgzwrqrve1bw7r81n4ktuajcudc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64998ad2757563ca37f7e7df

**Comment by Kingdutch** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3439-lbpxakgzwrqrve1bw7r81n4ktuajcudc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64998ad2757563ca37f7e7df

**Comment by Kingdutch** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3439-lbpxakgzwrqrve1bw7r81n4ktuajcudc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64998ad2757563ca37f7e7df

**Comment by Kingdutch** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3439-lbpxakgzwrqrve1bw7r81n4ktuajcudc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64998ad2757563ca37f7e7df

**Comment by Kingdutch** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3439-lbpxakgzwrqrve1bw7r81n4ktuajcudc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64998ad2757563ca37f7e7df

**Comment by Kingdutch** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3439-lbpxakgzwrqrve1bw7r81n4ktuajcudc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64998ad2757563ca37f7e7df

**Comment by Kingdutch** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3439-lbpxakgzwrqrve1bw7r81n4ktuajcudc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64998ad2757563ca37f7e7df

**Comment by ronaldtebrake** [comment]
Would be great if you could explain the changes you made, especially for the UUID a bit in here so we have some reference what occurred and why you fixed it this way. I feel like it's great knowledge transfer for others 😊

File: modules/social_features/social_core/social_core.install [hunk 3]
```diff
@@ -1561,10 +1572,18 @@ function social_core_update_11301(): void {
  * Add a new view mode for nodes - Email card.
  */
 function social_core_update_11401(): void {
-  $config_storage = \Drupal::service('config.storage');
-  $config_path = \Drupal::service('extension.list.module')->getPath('social_core') . '/config/static';
-  $source = new FileStorage($config_path);
-  $config_storage->write('core.entity_view_mode.node.email_card', (array) $source->read('core.entity_view_mode.node.email_card_11401'));
+
```
Comment (ronaldtebrake) [inline modules/social_features/social_core/social_core.install:1575]: think we can remove the excess whitespace (also on the other places)

**Comment by zanvidmar** [comment]
We can see that if `Xss::filter`'s first argument is NULL, it is now changed to an empty string at
https://github.com/goalgorilla/open_social/pull/3439/files#diff-032a2e5768d4846a91383aed3351d4534dc8469d33d0e949faafbfdbc27150dcR33

As `Xss::filter` requires first argument to be string, we can improve this here as well
https://github.com/goalgorilla/open_social/blob/94f607060c73fb81662e50e1fdbf5a342f538736/modules/custom/improved_theme_settings/improved_theme_settings.module#L255

**Comment by Kingdutch** [comment]
[Tugboat](https://www.tugboatqa.com) has finished building the preview for this pull request!

Link:
- https://pr3439-lbpxakgzwrqrve1bw7r81n4ktuajcudc.tugboatqa.com


Dashboard:
- https://dashboard.tugboatqa.com/64998ad2757563ca37f7e7df

**Review by zanvidmar — APPROVED** [review]

**Pull request merged**